packages/chat-stateful-client/src/StatefulChatClient.ts

```diff
@@ -13,6 +13,10 @@ export interface StatefulChatClient extends ChatClient {
   getState(): ChatClientState;
   onStateChange(handler: (state: ChatClientState) => void): void;
   offStateChange(handler: (state: ChatClientState) => void): void;
+  /**
+   * Clear all errors in the state for provided target methods.
+   */
+  clearErrors(targets: ChatErrorTargets[]): void;
```

Comment:
Hmm.. We are adding more apis to the stateful client (ideally it should be state related), but it seems to me that it is the only way to clear error. I am thinking about something like this in stateful client:

Do we have to expose the ability for stateful user to clear the lastErrors array? Is it possible that we don't clear the last error, but contoso choose whether to show it or not in their contoso logic/composite logic(Like a boolean value named 'hidden' directly into message bar)?

@DominikMe would be cool if you have a smart idea here

Reply:
We need _some_ way to clear old errors. Unlike chat messages, list of participants etc, the backing services don't have a store of encountered errors, so we aren't caching information from the backing services. So the clearing mechanism can't be -- clear errors in the backing services and the stateful cache gets updated automatically.

If we didn't want to store errors from stateful methods in the stateful client, we'd need a separate store just for errors, but that would be less ergonomic.

Reply:
Adding `clearErrors` feels wrong to me. Stateful clients should not expand on the base client API besides adding very few state specific functions.

Have we considered an `applyState(newState)` function instead? 

We can separately export helper methods like `clearErrors(state: ChatClientState): ChatClientState` which modify the state with immerJs to make it easier for the developer. That would be a similar pattern to DraftJs' `Modifier` for their Editor `ContentState`.
https://draftjs.org/docs/api-reference-modifier

Reply:
This is a great idea. 
Filed https://skype.visualstudio.com/SPOOL/_sprints/taskboard/Web%20UI/SPOOL/CY2021-Q3/Sprint%2041%20(Ends%20July%2025)?workitem=2519191 to iterate on this API on the top of the stack.